operator/src/test/java/oracle/kubernetes/operator/helpers/JobHelperTest.java

```diff
@@ -441,6 +444,55 @@ public void introspectorPodStartupWithNullAdminUsernamePasswordEnvVarValues() {
         getConfiguredDomainSpec(domainConfigurator).getEnv(), hasEnvVar("item1", RAW_VALUE_1));
   }
 
+  @Test
+  public void whenDomainHasAdditionalVolumesWithReservedVariables_createIntrospectorPodWithSubstitutions() {
+    configureDomain()
+        .withAdditionalVolumeMount("volume2", "/source-$(DOMAIN_UID)");
+    runCreateJob();
+    assertThat(
+        job.getSpec().getTemplate().getSpec().getContainers().get(0).getVolumeMounts(),
+        hasVolumeMount("volume2", "/source-" + UID));
+  }
+
+  @Test
+  public void whenDomainHasAdditionalVolumesWithCustomVariables_createIntrospectorPodWithSubstitutions() {
+    resourceLookup.defineResource(SECRET_NAME, KubernetesResourceType.Secret, NS);
+    resourceLookup.defineResource(OVERRIDES_CM_NAME_MODEL, KubernetesResourceType.ConfigMap, NS);
+    resourceLookup.defineResource(OVERRIDES_CM_NAME_IMAGE, KubernetesResourceType.ConfigMap, NS);
+
+    configureDomain()
+        .withEnvironmentVariable(ENV_NAME1, GOOD_MY_ENV_VALUE)
+        .withWebLogicCredentialsSecret(SECRET_NAME, null)
+        .withAdditionalVolume("volume1", VOLUME_PATH_1)
+        .withAdditionalVolumeMount("volume1", VOLUME_MOUNT_PATH_1);
+
+    runCreateJob();
+
+    assertThat(job.getSpec().getTemplate().getSpec().getContainers().get(0).getVolumeMounts(),
+            hasVolumeMount("volume1", END_VOLUME_MOUNT_PATH_1));
+  }
+
+  @Test
+  public void whenDomainHasAdditionalVolumesWithCustomVariablesInvalidValue_reportValidationError() {
```

Comment:
Should this test be named ..._jobNotCreated rather than ..._reportValidationError?

Reply:
sure. thanks!